samza-core/src/main/java/org/apache/samza/clustermanager/ResourceRequestState.java

```diff
@@ -205,6 +202,22 @@ public void updateStateAfterAssignment(SamzaResourceRequest request, String assi
     }
   }
 
+  /**
+   * Sends the {@link SamzaResourceRequest}s in the delayed requests queue that have expired.
+   * @return number of delayed requests sent.
+   */
+  public int sendExpiredDelayedResourceRequests() {
```

Comment:
Could we use another name, e.g., `sendPendingDelayedRequests`
In this code "expired resource requests" is also used to refer to resource requests which were never answered with a resource by the RM, so could confuse a new dev looking at this.